internal/fs/fs.go

```diff
@@ -143,6 +142,27 @@ type ServerConfig struct {
 	Notifier *fuse.Notifier
 }
 
+// newBufferedReadWorkerPool creates and starts a new worker pool for buffered reads.
+// The number of workers is determined based on the number of available CPUs.
+func newBufferedReadWorkerPool() (workerpool.WorkerPool, error) {
```

Comment:
Please move this constructor also in the static_worker_pool.go.

You can name it like - NewStaticWorkerPoolForCurrentCPU() (Pool, err).